### PR TITLE
Obfuscates pocket slots from the Strip Menu

### PR DESCRIPTION
**Pull request by keronshb**
## About the PR 

Hides pocket items from strip menu changes.

It is a bit hardcoded right now and the only other example of this in code hardcodes it something similar. I'd like to try to not hardcode it, but I'm having issues getting to that point.

**Screenshots**

image

**Changelog**

:cl:
- add: Adds obfuscation to pocket items so players can't check the strip menu to see what they have.

### CODE REVIEW

File: Content.Client/Inventory/StrippableBoundUserInterface.cs [hunk 1]
```diff
@@ -16,6 +16,7 @@ public sealed class StrippableBoundUserInterface : BoundUserInterface
         public Dictionary<(string ID, string Name), string>? Inventory { get; private set; }
         public Dictionary<string, string>? Hands { get; private set; }
         public Dictionary<EntityUid, string>? Handcuffs { get; private set; }
+        public string Occupied { get; } = "Occupied";
```
Comment (mirrorcult) [inline Content.Client/Inventory/StrippableBoundUserInterface.cs:19]: localize

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Comment by mirrorcult** [comment]
you could maybe have a datafield bool on the slot for whether it should be 'hidden'? and then check that instead of pocket1/pocket2

**Comment by keronshb** [comment]
> you could maybe have a datafield bool on the slot for whether it should be 'hidden'? and then check that instead of pocket1/pocket2

Oh that's a good idea, but in this instance how would I check for that? I know I can throw it on the slot like how it's done for the strip delay, but here it's seemingly not going through the inventory system the same way it would on the strip delay PR if that makes sense?

Like I tried to understand how it grabs inventory, but following in breakpoints or in code hasn't exactly been helpful on trying to figure out how it's grabbed.

For example if I made it I can't just do slot.IsHidden because even if I added it to the inventory template, it's not grabbing it from that.

I would add a bool to the dictionary so then it would have 3 variables? But again even if I followed up to the files that I know it's in (StrippableBounderUserInterface, SharedStrippableComponent) I just don't understand how it exactly grabs the data and how to work that in.

**Review by PaulRitter — CHANGES_REQUESTED** [review]
this should definitely be a bool. you'd need to network that bool using the strippingboundinterfacestate

**Comment by keronshb** [comment]
> this should definitely be a bool. you'd need to network that bool using the strippingboundinterfacestate

I do not understand how this works. If you can give me an example I could try, but I do not know what to add here or how to grab the information.

File: Content.Client/Inventory/StrippableBoundUserInterface.cs [hunk 2]
```diff
@@ -16,6 +16,7 @@ public sealed class StrippableBoundUserInterface : BoundUserInterface
         public Dictionary<(string ID, string Name), string>? Inventory { get; private set; }
         public Dictionary<string, string>? Hands { get; private set; }
         public Dictionary<EntityUid, string>? Handcuffs { get; private set; }
+        public string Occupied { get; } = Loc.GetString("strippable-bound-user-interface-stripping-menu-obfuscate");
```
Comment (ElectroJr) [inline Content.Client/Inventory/StrippableBoundUserInterface.cs:19]: `static`

**Comment by ElectroJr** [comment]
It looks like the easiest way to do this would be to just update the server's `StrippableSystem.SendUpdate()`, which is where it gets the slot & the name of the item. Just check if `slot.StripHidden` or whatever the bool will be called there and obfuscate the name of the item if needed

**Comment by keronshb** [comment]
> It looks like the easiest way to do this would be to just update the server's `StrippableSystem.SendUpdate()`, which is where it gets the slot & the name of the item. Just check if `slot.StripHidden` or whatever the bool will be called there and obfuscate the name of the item if needed

Got it. This latest push should fix it.

**Pull request merged**